lua/lspconfig/server_configurations/zls.lua

```diff
@@ -2,7 +2,7 @@ local util = require 'lspconfig.util'
 
 return {
   default_config = {
-    cmd = { 'zls' },
+    cmd = { 'zls', '--config-path', '"zls.json"' },
```

Comment:
why we don't check zls.json first build a cmd

Reply:
We don't really have to, since `zls` goes back to using global config if a specified config file does not exists, but we can.

Would you rather have `cmd` as a function that checks if `zls.json` exists in workspace and and then generates command based on that?

Reply:
This is fine if the server can fallback. Just wondering if it might bother some newbies.

Reply:
I tried looking into how to provide a function for `default_config.cmd`, but couldn't find it, does it always has to be a table?

Reply:
nope it can be a function.

Reply:
I'm struggling a bit here.
Tried providing a function instead of a table and printed the arguments, which got me:
```lua
{
  notification = <function 1>,
  on_error = <function 2>,
  on_exit = <function 3>,
  server_request = <function 4>
}
```
I then tried looking for these function names in source, but got unrelated stuff or in case of `server_request` even no results at all.
Tried looking through docs again, but they only seem to contain an example for table as well:
```
- {cmd} `list[string]`

  a list where each entry corresponds to the blankspace delimited part of
  the command that launches the server. The first entry is the binary used
  to run the language server. Additional entries are passed as arguments.

  The equivalent `cmd` for: 
 
  foo --bar baz
 
  is:
 
  {'foo', '--bar', 'baz'}
```
So not sure how to do this.
Or should I use `on_setup` hook instead?
Could you give some pointers on how this could be implemented?

Reply:
```lua
 cmd = function() xxxxxx end
```

Reply:
Added a check for `zls.json` first